Dietpi-Software | microblog.pub

mtekman opened the pull request:
Add [microblog.pub](https://microblog.pub/) , an activitypub powered microblog.

_Requirements:_

It requires Python3.10, so we create pyenv, and we clone the repo to the directory `~/microblog`

_Caveats_:

The installation is supposedly interactive, since it requires the user to enter a domain name, an admin password, but I can't make these interactive without significantly extending the app to generate the right hashes for all of the above.

Instead, it installs and starts default config, and then the user needs to re-configure it using a `microblog_pub configure` script, which loads the environment, and brings up the app's own configuration wizard.

mtekman committed:
add microblog-pub

mtekman committed:
shellfix

mtekman committed:
shellfix?

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -1571,6 +1571,12 @@ Available commands:
 		# - RISC-V: Missing package
 		aSOFTWARE_AVAIL_G_HW_ARCH[$software_id,11]=0
+		#-----------------
+		software_id=206
```
You can now use `software_id=16`, which became free with DietPi v8.15.

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -1571,6 +1571,12 @@ Available commands:
+		software_id=206
+		aSOFTWARE_NAME[$software_id]='microblog.pub'
+		aSOFTWARE_DESC[$software_id]='A self-hosted, single-user, ActivityPub powered microblog.'
+		aSOFTWARE_CATX[$software_id]=19
```
This fits much better to category 6 (Social & Search) where WordPress and phpBB are located.
```suggestion
		aSOFTWARE_CATX[$software_id]=6
```
Please also move this code blog up there.

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -1571,6 +1571,12 @@ Available commands:
+		aSOFTWARE_NAME[$software_id]='microblog.pub'
+		aSOFTWARE_DESC[$software_id]='A self-hosted, single-user, ActivityPub powered microblog.'
+		aSOFTWARE_CATX[$software_id]=19
+		aSOFTWARE_DOCS[$software_id]='https://docs.microblog.pub/'
```
Fine for now, but in the end we want to add a docs block to our docs and reference it here: https://dietpi.com/docs/software/social/

MichaIng replied:
Adding Git as dependency:
```suggestion
		aSOFTWARE_DOCS[$software_id]='https://docs.microblog.pub/'
		aSOFTWARE_DEPS[$software_id]='17'
```

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -4199,6 +4205,111 @@ _EOF_
 
+		if To_Install 206 # microblog.pub
+		then
+			## 1. Create a python environment py3.10 via pyenc
```
```suggestion
			## 1. Create a python environment py3.10 via pyenv
```
Does it have so many versioned dependencies that a `pyenv` makes sense? This always means a lengthy compiling step.

mtekman replied:
It has quite a few poetry dependencies. Is there an easier way without having to compile?

MichaIng replied:
Ah, I forgot about the minimum Python 3.10 requirement, so no way around then on Debian Buster and Bullseye. I guess on Debian Bookworm it could be skipped, but we can add this at a later time when the pyenv based implementation has been polished.

Suggestion about the little typo stays valid 😉.

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -4199,6 +4205,111 @@ _EOF_
+		then
+			## 1. Create a python environment py3.10 via pyenc
+			##	  see: https://python-poetry.org/docs/managing-environments/
+			local micro_pyenv_dir="/home/dietpi/.pyenv_microblog"
```
Please do not use the `dietpi` user, which is about to be removed. System services shall run as their dedicated service user.
```suggestion
			local micro_pyenv_dir='/opt/microblog-pub/.pyenv'
```

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -4199,6 +4205,111 @@ _EOF_
+## - poetry run inv migrate-db
+poetry env info | grep -oP '(?<=^Path:\s).*/virtualenvs/.*' | xargs echo > $micro_virtualenv
+"
+			if [[ $(cat "$micro_virtualenv") = "" ]]; then
```
```suggestion
			if [[ ! -s $micro_virtualenv ]]; then
```

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -4199,6 +4205,111 @@ _EOF_
+WantedBy=multi-user.target
+_EOF_
+
+			G_EXEC systemctl enable --now microblog-pub
```
```suggestion
```
Add the service to https://github.com/MichaIng/DietPi/blob/dev/dietpi/dietpi-services instead.

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -4199,6 +4205,111 @@ _EOF_
+_EOF_
+
+			G_EXEC systemctl enable --now microblog-pub
+			G_EXEC_DESC="In order to complete the microblog-pub installation, you need to run 'microblog_pub configure' in a new bash shell!"
```
```suggestion
			G_WHIP_MSG '[ INFO] microblog.pub installation finished
\nIn order to complete the setup, please run "microblog_pub configure" in a new bash.'
```

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -13153,6 +13264,23 @@ If no WireGuard (auto)start is included, but you require it, please do the follo
+			then
+				G_EXEC systemctl disable --now microblog-pub
+				G_EXEC rm "$micro_systemd"
+			fi
```
Remove possible service overrides, e.g. from `dietpi-services`, as well.
```suggestion
			fi
			[[ -d $micro_systemd.d ]] && G_EXEC rm -R "$micro_systemd.d"
```

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -4199,6 +4205,111 @@ _EOF_
+			local micro_pyenv_dir="/home/dietpi/.pyenv_microblog"
+			local micro_pyenv_activate_file="$micro_pyenv_dir/pyenv-activate.sh"
+			local micro_python_version='3.10.7'
+			local micro_dir="/home/dietpi/microblog"
```
To align with other installs, config/data dir to `/mnt/dietpi_userdata`:
```suggestion
			local micro_dir='/mnt/dietpi_userdata/microblog-pub/.pyenv'
```

mtekman committed:
initial feedback changes, untested

MichaIng committed:
v8.16

- DietPi-Software | microblog.pub: Bump Python to 3.10.9, align naming of user, service and all directories, move virtualenv path file into pyenv dir, install pyenv, Python and Poetry in one step, update uninstall code to benefit from new Remove_Service function

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -4136,6 +4143,103 @@ _EOF_
+User=$micro_name
+SyslogIdentifier=microblog.pub
+WorkingDirectory=$micro_data_dir
+Environment=VENV_DIR=$micro_virtualenv
```
Shouldn't it actually be a directory path here instead of a path to a file which contains the directory path?

MichaIng replied:
I broke it myself, just fixed

MichaIng committed:
v8.16

- DietPi-Software | microblog.pub: Add build dependencies, fix source command for POSIX shell and abort activation and pyenv install script on first error

MichaIng committed:
v8.16

- DietPi-Software | microblog.pub: Fix config script by doing everything as run user

MichaIng committed:
v8.16

- DietPi-Software | microblog.pub: Fix VENV_DIR variable

MichaIng committed:
v8.16

- DietPi-Software | microblog.pub: Minor

MichaIng committed:
v8.16

- DietPi-Software | microblog.pub: Enable remote access on port 8007 and disable pip cache
- DietPi-Software | Home Assistant: Disable pip cache

MichaIng committed:
v8.16

- DietPi-Software | microblog.pub: Add an info about adding the port to the domain when doing the "microblog-pub configure". This is needed since assets are requested will full URL based on the entered domain and a non-matching port would not only fail to load them but also be blocked by CSP.
